core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+package promwrapper
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_query_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Query() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "plugin"},
+	)
+	promObservation = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_observation_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Observation() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Report() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldAcceptFinalizedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_accept_finalized_report_duration",
```

Comment:
I see some other examples of `_second`, `_ms`, and `_time` - could we consider using one of those rather than introducing new suffix `_duration`?
```suggestion
			Name:    "ocr2_reporting_plugin_should_accept_finalized_report_time",
```
It seems like the raw ns value is used for `_time` :shrug:

Reply:
yeah seems like we should use `_time` to be consistent with existing code :+1:

Reply:
There's some guidelines [here](https://prometheus.io/docs/practices/histograms/) on naming as well, that's from the prometheus project directly, though not sure if we're following those.

Reply:
@jinhoonbang so we're going to switch these to be `_time` rather than `_duration` right?

Reply:
Out of scope: I'm wondering again if we should move all of our prom metrics in to one package so we can see them all together... :thinking: